server/channels/app/user.go

```diff
@@ -2817,3 +2818,53 @@ func (a *App) UserIsFirstAdmin(user *model.User) bool {
 
 	return true
 }
+
+func (a *App) GetUsersForReporting(
+	sortColumn string,
+	sortDesc bool,
+	pageSize int,
+	lastSortColumnValue string,
+	lastUserId string,
+	startAt int64,
+	endAt int64,
+) ([]*model.UserReport, *model.AppError) {
+	// Don't allow fetching more than 100 users at a time from the normal query endpoint
+	if pageSize <= 0 || pageSize > 100 {
+		return nil, model.NewAppError("GetUsersForReporting", "app.user.get_users_for_reporting.invalid_page_size", nil, "", http.StatusBadRequest)
+	}
+
+	// Validate date range
+	if endAt > 0 && startAt > endAt {
+		return nil, model.NewAppError("GetUsersForReporting", "app.user.get_users_for_reporting.bad_date_range", nil, "", http.StatusBadRequest)
+	}
+
+	return a.getUserReport(sortColumn, sortDesc, pageSize, lastSortColumnValue, lastUserId, startAt, endAt)
+}
+
+func (a *App) getUserReport(
+	sortColumn string,
+	sortDesc bool,
+	pageSize int,
+	lastSortColumnValue string,
+	lastUserId string,
+	startAt int64,
+	endAt int64,
+) ([]*model.UserReport, *model.AppError) {
+	// Validate against the columns we allow sorting for
+	if !pUtils.Contains(model.UserReportSortColumns, sortColumn) {
+		return nil, model.NewAppError("GetUsersForReporting", "app.user.get_user_report.invalid_sort_column", nil, "", http.StatusBadRequest)
+	}
+
+	userReportQuery, err := a.Srv().Store().User().GetUserReport(sortColumn, sortDesc, pageSize, lastSortColumnValue, lastUserId, startAt, endAt)
+	if err != nil {
+		return nil, model.NewAppError("GetUsersForReporting", "app.user.get_user_report.store_error", nil, "", http.StatusInternalServerError).Wrap(err)
+	}
+
+	userReports := []*model.UserReport{}
+	for _, user := range userReportQuery {
+		report := user.ToReport()
+		userReports = append(userReports, &report)
+	}
```

Comment:
Could we just return reference in the `user.ToReport()`? Then we wouldn't need to get the reference in this loop. 
```suggestion
	userReports := make([]*model.UserReport, len(userReportQuery)
	for i, user := range userReportQuery {
		report := user.ToReport()
		userReports[i] = &report
	}
```

Reply:
Sure, seems like a good change.